spec/lib/manageiq/active_record_connector_spec.rb

```diff
@@ -0,0 +1,108 @@
+require 'manageiq/active_record_connector'
+
+describe ManageIQ::ActiveRecordConnector do
+  def without_rails(rb_cmd)
+    file = Rails.root.join("lib", "manageiq", "active_record_connector")
+    `#{Gem.ruby} -e "require '#{file}'; #{rb_cmd}"`
+  end
+
+  describe ".connection_exists" do
+    it "returns true when a connection is already established" do
+      expect(described_class.connection_exists?).to be true
+    end
+
+    it "returns false when a connection does already not exist" do
+      cmd = "print #{described_class}.connection_exists?"
+      expect(without_rails(cmd)).to eq("false")
+    end
+  end
+
+  describe ".establish_connection_if_needed" do
+    # Save off existing setup
+    let!(:config)     { ActiveRecord::Base.connection_config }
+    let!(:connection) { ActiveRecord::Base.connection }
+    let!(:logger)     { ActiveRecord::Base.logger }
+
+    it "doesn't create a new connection/logger if it doesn't need to" do
+      described_class.establish_connection_if_needed(config, logger.filename)
+
+      expect(ActiveRecord::Base.connection).to eq(connection)
+      expect(ActiveRecord::Base.logger).to     eq(logger)
+    end
+
+    it "creates a new connection as needed" do
+      cmd_establish_connection   = "#{described_class}.establish_connection_if_needed(#{config.inspect.gsub(/(")/, "\\\\\\1")}, '#{logger.filename.to_s}')"
+      cmd_connection_exists      = "puts #{described_class}.connection_exists?"
+      # Not sure why this is a pain in the butt... but whatever...
```

Comment:
Forgot this was in tests, but yeah, it was because `util/vmdb-logger.rb` exposes this stuff that I was able to make use `.filename` directly, but doesn't exists on a vanilla logger (which is used in the `without_rails` function.  #funtimes